---
title: "EEP_cDNA_pDNA_Preprocessing_MMA20230802"
author: "MMA"
date: '2023-08-03'
output: html_document
---

```{r knitr setup, include=FALSE}
prefix_lab <- 'EEP_iPCR_ecn_dattero_V2'
knitr::opts_chunk$set(echo = TRUE,
                      include = TRUE,
                      message = FALSE, 
                      warning = FALSE, 
                      cache = TRUE, 
                      cache.path = paste0('cache/', prefix_lab, '/'),
                      fig.path   = paste0('figures/', prefix_lab, '/'),
                      fig.align = 'center', 
                      fig.width = 5, 
                      fig.height = 5,
                      knitr.table.format = 'markdown')
```

```{r libs and fun, include = FALSE, tidy = FALSE}
library(tidyverse)
library(here)
library(magrittr)
library(reshape2)
library(knitr)
library(gridExtra)
library(pander)
library(cowplot)
library(circlize)
library(datteRo)
library(ggplot2)
library(GGally)
library(scales)
library(heatmaply)
library(ggpubr)
```

---

# Intro

In this document we preprocess sequencing data into activity and boost indices tables.

iPCR data from each EEP library (matches barcode to EE insert identity) is merged with pDNA and cDNA barcode count data. Clashing barcodes are removed, barcodes with less than 8 pDNA counts are filtered out and EE insert with less than 5 barcodes are removed. Then activities are calculated as cDNA to pDNA ratios per barcode. Activities are averaged across 3 biological replicates using a geometric mean. Then control-control combinations are used to calculate the baseline of each promoter to calculate boost indices as the log2 ratio between the the activity of each combination and the baseline. Single enhancer effects are calculated as the median activity across all enhancer-control combinations of a given enhancer.

The iPCR and count data was generated using the snakemake pipeline available in this repository.

Some of the functions used are custom functions compiled in the ´datteRo´ package available in this repository.

# Load iPCRs

```{r}
# custom load iPCR func
read_ipcr_EEP <- function(fn, delim = '\t') {

  # set column names
  col_names <- c('barcode', 'frag1', 'strand1', 'frag2', 'strand2', 'ipcr_counts', 'cluster_id')

  # read ipcr data as tibble
  tib <- readr::read_delim(fn, delim = delim, col_names = col_names)

  # strip DAT library name
  # pad fragment id with 0's
  # add unique fragment identifier
  tib %>%
    mutate(prefix1 = str_sub(frag1, 1, 1),
           suffix1 = str_remove(frag1, ".*_"),
           frag1 = str_remove(frag1, '_.*'),
           frag1 = str_remove(frag1, '[EP]'),
           frag1 = str_pad(frag1, width = 3, pad = '0'),
           frag1 = paste0(suffix1 ,'_',prefix1, frag1),
           prefix2 = str_sub(frag2, 1, 1),
           suffix2 = str_remove(frag2, ".*_"),
           frag2 = str_remove(frag2, '_.*'),
           frag2 = str_remove(frag2, '[EP]'),
           frag2 = str_pad(frag2, width = 3, pad = '0'),
           frag2 = paste0(suffix2 ,'_',prefix2, frag2),
           id = paste0(frag1, strand1, frag2, strand2))
}

```

```{r process libs}
# load EEP libraries
tib_EEP2_Sox2 <- as.tibble(read_ipcr_EEP('/DATA/projects/epmatrix/mouse/TripleCombs/ipcr/Results/Sox2_EEP_iPCR_unique_barcodes_frag_id_merged.tsv'))
tib_EEP2_Klf2.merged <- as.tibble(read_ipcr_EEP('/DATA/projects/epmatrix/mouse/TripleCombs/ipcr/Results/Klf2.merged_EEP_iPCR_unique_barcodes_frag_id_merged.tsv'))
tib_EEP2_Otx2 <- as.tibble(read_ipcr_EEP('/DATA/projects/epmatrix/mouse/TripleCombs/ipcr/Results/Otx2_EEP_iPCR_unique_barcodes_frag_id_merged.tsv'))
tib_EEP2_FGF5 <- as.tibble(read_ipcr_EEP('/DATA/projects/epmatrix/mouse/TripleCombs/ipcr/Results/FGF5_EEP_iPCR_unique_barcodes_frag_id_merged.tsv'))
tib_EEP2_Lefty1 <- as.tibble(read_ipcr_EEP('/DATA/projects/epmatrix/mouse/TripleCombs/ipcr/Results/Lefty1_EEP_iPCR_unique_barcodes_frag_id_merged.tsv'))
tib_EEP2_Nanog <- as.tibble(read_ipcr_EEP('/DATA/projects/epmatrix/mouse/TripleCombs/ipcr/Results/Nanog_EEP_iPCR_unique_barcodes_frag_id_merged.tsv'))
tib_EEP2_P14 <- as.tibble(read_ipcr_EEP('/DATA/projects/epmatrix/mouse/TripleCombs/ipcr/Results/P14_EEP_iPCR_unique_barcodes_frag_id_merged.tsv'))
tib_EEP2_Tbx3 <- as.tibble(read_ipcr_EEP('/DATA/projects/epmatrix/mouse/TripleCombs/ipcr/Results/Tbx3_EEP_iPCR_unique_barcodes_frag_id_merged.tsv'))

# load spike-in libraries

tib_Spike_in_P1 <- as.tibble(read_ipcr('~/epmatrix/mouse/Split/ipcr/Results/ipcr_E62_2_P-E002_Split_unique_barcodes_frag_id_merged.tsv'))
tib_Spike_in_P16 <- as.tibble(read_ipcr('~/epmatrix/mouse/Split/ipcr/Results/ipcr_E62_6_P-E006_Split_unique_barcodes_frag_id_merged.tsv'))

```

```{r}
## correct suffixes Spike ins
tib_Spike_in_P1<-tib_Spike_in_P1%>% mutate(prefix1="SP1",prefix2="SP1", suffix1="SP1", suffix2="SP1", frag1="SpikeIn1", frag2="SpikeIn1", strand1= "*", strand2= "*", id= paste0(frag1, strand1, frag2, strand2))
tib_Spike_in_P16<-tib_Spike_in_P16%>% mutate(prefix1="SP16",prefix2="SP16", suffix1="SP16", suffix2="SP16", frag1="SpikeIn16", frag2="SpikeIn16", strand1= "*", strand2= "*", id= paste0(frag1, strand1, frag2, strand2))
```


# iPCR preprocesssing


```{r}
#Total number of combinations tested
tib_EEP2_Sox2%>%
  group_by(id)%>%
  summarise()%>%
  nrow()

tib_EEP2_Klf2.merged%>%
  group_by(id)%>%
  summarise()%>%
  nrow()

tib_EEP2_Otx2%>%
  group_by(id)%>%
  summarise()%>%
  nrow()

tib_EEP2_FGF5%>%
  group_by(id)%>%
  summarise()%>%
  nrow()

tib_EEP2_Lefty1%>%
  group_by(id)%>%
  summarise()%>%
  nrow()

tib_EEP2_Nanog%>%
  group_by(id)%>%
  summarise()%>%
  nrow()

tib_EEP2_P14%>%
  group_by(id)%>%
  summarise()%>%
  nrow()

tib_EEP2_Tbx3%>%
  group_by(id)%>%
  summarise()%>%
  nrow()

```

## Correct data structure
 

In order to make comparisons easier between the Datasets we process each library separatedly. We keep the the same structure as the previous EP libraries. For this purpose, we will keep the terminology fragment 1 and fragment 2 but change its meaning. Being in this case fragment 1 the firs enhancer and fragment 2 the second enhancer in the enhancer positon. Then we add a column which indicates the promoter of each library. In this process we also correct "eempty" fragments to "empty", and prefix "e" for noF


```{r}
#Sox2
tib_EEP2_Sox2_renamed <- tib_EEP2_Sox2  %>% mutate(frag1 = ifelse(frag1=="eempty", "empty", frag1)) %>% mutate(prefix1 = ifelse(prefix1=="e", "NoF", prefix1)) %>% mutate(Promoter="P1_SOX2") %>% mutate(id =paste0(frag1, strand1, frag2, strand2, Promoter)) %>% mutate(id1 =paste0(frag1, strand1), id2 =paste0(frag2, strand2)) %>% mutate(library="Sox2")

#Klf2
tib_EEP2_Klf2.merged_renamed <- tib_EEP2_Klf2.merged  %>% mutate(frag1 = ifelse(frag1=="eempty", "empty", frag1)) %>% mutate(prefix1 = ifelse(prefix1=="e", "NoF", prefix1)) %>% mutate(Promoter="P2_KLF2") %>% mutate(id =paste0(frag1, strand1, frag2, strand2, Promoter)) %>% mutate(id1 =paste0(frag1, strand1), id2 =paste0(frag2, strand2)) %>% mutate(library="Klf2")

#Otx2
tib_EEP2_Otx2_renamed <- tib_EEP2_Otx2  %>% mutate(frag1 = ifelse(frag1=="eempty", "empty", frag1)) %>% mutate(prefix1 = ifelse(prefix1=="e", "NoF", prefix1)) %>% mutate(Promoter="P3_OTX2") %>% mutate(id =paste0(frag1, strand1, frag2, strand2, Promoter)) %>% mutate(id1 =paste0(frag1, strand1), id2 =paste0(frag2, strand2)) %>% mutate(library="Otx2")

#FGF5
tib_EEP2_FGF5_renamed <- tib_EEP2_FGF5  %>% mutate(frag1 = ifelse(frag1=="eempty", "empty", frag1)) %>% mutate(prefix1 = ifelse(prefix1=="e", "NoF", prefix1)) %>% mutate(Promoter="P4_FGF5") %>% mutate(id =paste0(frag1, strand1, frag2, strand2, Promoter)) %>% mutate(id1 =paste0(frag1, strand1), id2 =paste0(frag2, strand2)) %>% mutate(library="FGF5")

#Lefty1
tib_EEP2_Lefty1_renamed <- tib_EEP2_Lefty1  %>% mutate(frag1 = ifelse(frag1=="eempty", "empty", frag1)) %>% mutate(prefix1 = ifelse(prefix1=="e", "NoF", prefix1)) %>% mutate(Promoter="P5_Lefty1") %>% mutate(id =paste0(frag1, strand1, frag2, strand2, Promoter)) %>% mutate(id1 =paste0(frag1, strand1), id2 =paste0(frag2, strand2)) %>% mutate(library="Lefty1")

#Nanog
tib_EEP2_Nanog_renamed <- tib_EEP2_Nanog  %>% mutate(frag1 = ifelse(frag1=="eempty", "empty", frag1)) %>% mutate(prefix1 = ifelse(prefix1=="e", "NoF", prefix1)) %>% mutate(Promoter="P6_Nanog") %>% mutate(id =paste0(frag1, strand1, frag2, strand2, Promoter)) %>% mutate(id1 =paste0(frag1, strand1), id2 =paste0(frag2, strand2)) %>% mutate(library="Nanog")

#P14
tib_EEP2_P14_renamed <- tib_EEP2_P14  %>% mutate(frag1 = ifelse(frag1=="eempty", "empty", frag1)) %>% mutate(prefix1 = ifelse(prefix1=="e", "NoF", prefix1)) %>% mutate(Promoter="P7_Ap1m1") %>% mutate(id =paste0(frag1, strand1, frag2, strand2, Promoter)) %>% mutate(id1 =paste0(frag1, strand1), id2 =paste0(frag2, strand2)) %>% mutate(library="Ap1m1")

#Tbx3
tib_EEP2_Tbx3_renamed <- tib_EEP2_Tbx3  %>% mutate(frag1 = ifelse(frag1=="eempty", "empty", frag1)) %>% mutate(prefix1 = ifelse(prefix1=="e", "NoF", prefix1)) %>% mutate(Promoter="P8_Tbx3") %>% mutate(id =paste0(frag1, strand1, frag2, strand2, Promoter)) %>% mutate(id1 =paste0(frag1, strand1), id2 =paste0(frag2, strand2)) %>% mutate(library="Tbx3")

# Spike-in P1
tib_Spike_in_P1_renamed <- tib_Spike_in_P1  %>% mutate(frag1 = ifelse(frag1=="eempty", "empty", frag1)) %>% mutate(prefix1 = ifelse(prefix1=="e", "NoF", prefix1)) %>% mutate(Promoter="SP_Zfp822") %>% mutate(id =paste0(frag1, strand1, frag2, strand2, Promoter)) %>% mutate(id1 =paste0(frag1, strand1), id2 =paste0(frag2, strand2)) %>% mutate(library="Spike_in_P1")

# Spike-in P16_Klf2
tib_Spike_in_P16_renamed <- tib_Spike_in_P16  %>% mutate(frag1 = ifelse(frag1=="eempty", "empty", frag1)) %>% mutate(prefix1 = ifelse(prefix1=="e", "NoF", prefix1)) %>% mutate(Promoter="SP_Klf2") %>% mutate(id =paste0(frag1, strand1, frag2, strand2, Promoter)) %>% mutate(id1 =paste0(frag1, strand1), id2 =paste0(frag2, strand2)) %>% mutate(library="Spike_in_P16")

```

## Combine libs with Spike-in libs

We combine each library with the spike-in libs and then remove clashing barcodes.

```{r}
#Combine Sox2
tib_Split_EEP_Sox2_Combined <- bind_rows(tib_EEP2_Sox2_renamed, tib_Spike_in_P1_renamed, tib_Spike_in_P16_renamed)
#Combine Klf2
tib_Split_EEP_Klf2_Combined <- bind_rows(tib_EEP2_Klf2.merged_renamed, tib_Spike_in_P1_renamed, tib_Spike_in_P16_renamed)
#Combine Otx2
tib_Split_EEP_Otx2_Combined <- bind_rows(tib_EEP2_Otx2_renamed, tib_Spike_in_P1_renamed, tib_Spike_in_P16_renamed)
#Combine FGF5
tib_Split_EEP_FGF5_Combined <- bind_rows(tib_EEP2_FGF5_renamed, tib_Spike_in_P1_renamed, tib_Spike_in_P16_renamed)
#Combine Lefty1
tib_Split_EEP_Lefty1_Combined <- bind_rows(tib_EEP2_Lefty1_renamed, tib_Spike_in_P1_renamed, tib_Spike_in_P16_renamed)
#Combine Nanog
tib_Split_EEP_Nanog_Combined <- bind_rows(tib_EEP2_Nanog_renamed, tib_Spike_in_P1_renamed, tib_Spike_in_P16_renamed)
#Combine P14
tib_Split_EEP_P14_Combined <- bind_rows(tib_EEP2_P14_renamed, tib_Spike_in_P1_renamed, tib_Spike_in_P16_renamed)
#Combine Tbx3
tib_Split_EEP_Tbx3_Combined <- bind_rows(tib_EEP2_Tbx3_renamed, tib_Spike_in_P1_renamed, tib_Spike_in_P16_renamed)

#####Remove clashing barcodes

#Remove clashing barcodes Sox2
tib_Split_EEP_Sox2_Combined_Unique_PlusUnknw <-tib_Split_EEP_Sox2_Combined %>%nest(-barcode ) %>% mutate(n = map_dbl(data, nrow))%>%
  filter(n==1)%>%
  unnest()%>% 
  select(-n)

#Remove clashing barcodes Klf2
tib_Split_EEP_Klf2_Combined_Unique_PlusUnknw <-tib_Split_EEP_Klf2_Combined %>%nest(-barcode ) %>% mutate(n = map_dbl(data, nrow))%>%
  filter(n==1)%>%
  unnest()%>% 
  select(-n)

#Remove clashing barcodes Otx2
tib_Split_EEP_Otx2_Combined_Unique_PlusUnknw <-tib_Split_EEP_Otx2_Combined %>%nest(-barcode ) %>% mutate(n = map_dbl(data, nrow))%>%
  filter(n==1)%>%
  unnest()%>% 
  select(-n)

#Remove clashing barcodes FGF5
tib_Split_EEP_FGF5_Combined_Unique_PlusUnknw <-tib_Split_EEP_FGF5_Combined %>%nest(-barcode ) %>% mutate(n = map_dbl(data, nrow))%>%
  filter(n==1)%>%
  unnest()%>% 
  select(-n)

#Remove clashing barcodes Lefty1
tib_Split_EEP_Lefty1_Combined_Unique_PlusUnknw <-tib_Split_EEP_Lefty1_Combined %>%nest(-barcode ) %>% mutate(n = map_dbl(data, nrow))%>%
  filter(n==1)%>%
  unnest()%>% 
  select(-n)

#Remove clashing barcodes Nanog
tib_Split_EEP_Nanog_Combined_Unique_PlusUnknw <-tib_Split_EEP_Nanog_Combined %>%nest(-barcode ) %>% mutate(n = map_dbl(data, nrow))%>%
  filter(n==1)%>%
  unnest()%>% 
  select(-n)

#Remove clashing barcodes P14
tib_Split_EEP_P14_Combined_Unique_PlusUnknw <-tib_Split_EEP_P14_Combined %>%nest(-barcode ) %>% mutate(n = map_dbl(data, nrow))%>%
  filter(n==1)%>%
  unnest()%>% 
  select(-n)

#Remove clashing barcodes Tbx3
tib_Split_EEP_Tbx3_Combined_Unique_PlusUnknw <-tib_Split_EEP_Tbx3_Combined %>%nest(-barcode ) %>% mutate(n = map_dbl(data, nrow))%>%
  filter(n==1)%>%
  unnest()%>% 
  select(-n)
```
# clean environment

```{r}
remove(tib_EEP2_Sox2_renamed, tib_EEP2_Sox2, tib_EEP2_Klf2.merged_renamed, tib_EEP2_Klf2.merged, tib_EEP2_Otx2_renamed, tib_EEP2_Otx2,tib_EEP2_FGF5_renamed, tib_EEP2_FGF5, tib_EEP2_Lefty1_renamed,tib_EEP2_Lefty1,tib_EEP2_Nanog_renamed,tib_EEP2_Nanog,tib_EEP2_P14_renamed,tib_EEP2_P14,tib_EEP2_Tbx3_renamed,tib_EEP2_Tbx3, tib_Spike_in_P1_renamed, tib_Spike_in_P1, tib_Spike_in_P16_renamed, tib_Spike_in_P16 )
```


# Combining expression and copy number data
Combine processed iPCR data with cDNA and pDNA count data. 
### Sox2

```{r merge ipcr and ecn data Sox2}
# read expression data (cdna)
tib_expr_br1_Sox2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Sox2_cDNA_merged_BR1_barcode_counts.tsv', col_lab = 'cdna_counts_br1')
tib_expr_br2_Sox2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Sox2_cDNA_merged_BR2_barcode_counts.tsv', col_lab = 'cdna_counts_br2')
tib_expr_br3_Sox2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Sox2_cDNA_merged_BR3_barcode_counts.tsv', col_lab = 'cdna_counts_br3')
# read copy number data (pdna)
tib_cn_br1_Sox2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Sox2_pDNA_merged_barcode_counts.tsv', col_lab = 'pdna_counts_br1')

# merge datasets Sox2
tib_Split_EEP_Sox2_merged <- merge_ipcr_ecn(tib           = tib_Split_EEP_Sox2_Combined_Unique_PlusUnknw, 
                                   cn_tib_list   = list(tib_cn_br1_Sox2),
                                   expr_tib_list = list(tib_expr_br1_Sox2,tib_expr_br2_Sox2,tib_expr_br3_Sox2))


```

### Klf2

```{r merge ipcr and ecn data Klf2}
# read expression data (cdna)
tib_expr_br1_Klf2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Klf2_cDNA_merged_BR1_barcode_counts.tsv', col_lab = 'cdna_counts_br1')
tib_expr_br2_Klf2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Klf2_cDNA_merged_BR2_barcode_counts.tsv', col_lab = 'cdna_counts_br2')
tib_expr_br3_Klf2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Klf2_cDNA_merged_BR3_barcode_counts.tsv', col_lab = 'cdna_counts_br3')
# read copy number data (pdna)
tib_cn_br1_Klf2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Klf2_pDNA_merged_barcode_counts.tsv', col_lab = 'pdna_counts_br1')

# merge datasets Klf2
tib_Split_EEP_Klf2_merged <- merge_ipcr_ecn(tib           = tib_Split_EEP_Klf2_Combined_Unique_PlusUnknw, 
                                   cn_tib_list   = list(tib_cn_br1_Klf2),
                                   expr_tib_list = list(tib_expr_br1_Klf2,tib_expr_br2_Klf2,tib_expr_br3_Klf2))



```

### Otx2

```{r merge ipcr and ecn data Otx2}
# read expression data (cdna)
tib_expr_br1_Otx2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Otx2_cDNA_merged_BR1_barcode_counts.tsv', col_lab = 'cdna_counts_br1')
tib_expr_br2_Otx2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Otx2_cDNA_merged_BR2_barcode_counts.tsv', col_lab = 'cdna_counts_br2')
tib_expr_br3_Otx2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Otx2_cDNA_merged_BR3_barcode_counts.tsv', col_lab = 'cdna_counts_br3')
# read copy number data (pdna)
tib_cn_br1_Otx2 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Otx2_pDNA_merged_barcode_counts.tsv', col_lab = 'pdna_counts_br1')

# merge datasets Klf2
tib_Split_EEP_Otx2_merged <- merge_ipcr_ecn(tib           = tib_Split_EEP_Otx2_Combined_Unique_PlusUnknw, 
                                   cn_tib_list   = list(tib_cn_br1_Otx2),
                                   expr_tib_list = list(tib_expr_br1_Otx2,tib_expr_br2_Otx2,tib_expr_br3_Otx2))



```

### FGF5

```{r merge ipcr and ecn data FGF5}
# read expression data (cdna)
tib_expr_br1_FGF5 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_FGF5_cDNA_merged_BR1_barcode_counts.tsv', col_lab = 'cdna_counts_br1')
tib_expr_br2_FGF5 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_FGF5_cDNA_merged_BR2_barcode_counts.tsv', col_lab = 'cdna_counts_br2')
tib_expr_br3_FGF5 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_FGF5_cDNA_merged_BR3_barcode_counts.tsv', col_lab = 'cdna_counts_br3')
# read copy number data (pdna)
tib_cn_br1_FGF5 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_FGF5_pDNA_merged_barcode_counts.tsv', col_lab = 'pdna_counts_br1')

# merge datasets FGF5
tib_Split_EEP_FGF5_merged <- merge_ipcr_ecn(tib           = tib_Split_EEP_FGF5_Combined_Unique_PlusUnknw, 
                                   cn_tib_list   = list(tib_cn_br1_FGF5),
                                   expr_tib_list = list(tib_expr_br1_FGF5,tib_expr_br2_FGF5,tib_expr_br3_FGF5))



```

### Lefty1

```{r merge ipcr and ecn data Lefty1}
# read expression data (cdna)
tib_expr_br1_Lefty1 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Lefty1_cDNA_merged_BR1_barcode_counts.tsv', col_lab = 'cdna_counts_br1')
tib_expr_br2_Lefty1 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Lefty1_cDNA_merged_BR2_barcode_counts.tsv', col_lab = 'cdna_counts_br2')
tib_expr_br3_Lefty1 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Lefty1_cDNA_merged_BR3_barcode_counts.tsv', col_lab = 'cdna_counts_br3')
# read copy number data (pdna)
tib_cn_br1_Lefty1 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Lefty1_pDNA_merged_barcode_counts.tsv', col_lab = 'pdna_counts_br1')

# merge datasets FGF5
tib_Split_EEP_Lefty1_merged <- merge_ipcr_ecn(tib           = tib_Split_EEP_Lefty1_Combined_Unique_PlusUnknw, 
                                   cn_tib_list   = list(tib_cn_br1_Lefty1),
                                   expr_tib_list = list(tib_expr_br1_Lefty1,tib_expr_br2_Lefty1,tib_expr_br3_Lefty1))



```

### Nanog

```{r merge ipcr and ecn data Nanog}
# read expression data (cdna)
tib_expr_br1_Nanog <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Nanog_cDNA_merged_BR1_barcode_counts.tsv', col_lab = 'cdna_counts_br1')
tib_expr_br2_Nanog <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Nanog_cDNA_merged_BR2_barcode_counts.tsv', col_lab = 'cdna_counts_br2')
tib_expr_br3_Nanog <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Nanog_cDNA_merged_BR3_barcode_counts.tsv', col_lab = 'cdna_counts_br3')
# read copy number data (pdna)
tib_cn_br1_Nanog <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Nanog_pDNA_merged_barcode_counts.tsv', col_lab = 'pdna_counts_br1')

# merge datasets Nanog
tib_Split_EEP_Nanog_merged <- merge_ipcr_ecn(tib           = tib_Split_EEP_Nanog_Combined_Unique_PlusUnknw, 
                                   cn_tib_list   = list(tib_cn_br1_Nanog),
                                   expr_tib_list = list(tib_expr_br1_Nanog,tib_expr_br2_Nanog,tib_expr_br3_Nanog))



```

### P14

```{r merge ipcr and ecn data P14}
# read expression data (cdna)
tib_expr_br1_P14 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_P14_cDNA_merged_BR1_barcode_counts.tsv', col_lab = 'cdna_counts_br1')
tib_expr_br2_P14 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_P14_cDNA_merged_BR2_barcode_counts.tsv', col_lab = 'cdna_counts_br2')
tib_expr_br3_P14 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_P14_cDNA_merged_BR3_barcode_counts.tsv', col_lab = 'cdna_counts_br3')
# read copy number data (pdna)
tib_cn_br1_P14 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_P14_pDNA_merged_barcode_counts.tsv', col_lab = 'pdna_counts_br1')

# merge datasets Nanog
tib_Split_EEP_P14_merged <- merge_ipcr_ecn(tib           = tib_Split_EEP_P14_Combined_Unique_PlusUnknw, 
                                   cn_tib_list   = list(tib_cn_br1_P14),
                                   expr_tib_list = list(tib_expr_br1_P14,tib_expr_br2_P14,tib_expr_br3_P14))



```

### Tbx3

```{r merge ipcr and ecn data Tbx3}
# read expression data (cdna)
tib_expr_br1_Tbx3 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Tbx3_cDNA_merged_BR1_barcode_counts.tsv', col_lab = 'cdna_counts_br1')
tib_expr_br2_Tbx3 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Tbx3_cDNA_merged_BR2_barcode_counts.tsv', col_lab = 'cdna_counts_br2')
tib_expr_br3_Tbx3 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Tbx3_cDNA_merged_BR3_barcode_counts.tsv', col_lab = 'cdna_counts_br3')
# read copy number data (pdna)
tib_cn_br1_Tbx3 <- read_ecn('/DATA/projects/epmatrix/mouse/TripleCombs/ecn/Results/EEP_Tbx3_pDNA_merged_barcode_counts.tsv', col_lab = 'pdna_counts_br1')

# merge datasets Tbx3
tib_Split_EEP_Tbx3_merged <- merge_ipcr_ecn(tib           = tib_Split_EEP_Tbx3_Combined_Unique_PlusUnknw, 
                                   cn_tib_list   = list(tib_cn_br1_Tbx3),
                                   expr_tib_list = list(tib_expr_br1_Tbx3,tib_expr_br2_Tbx3,tib_expr_br3_Tbx3))



```



### Barcodes lost

```{r}
get_barcode_summary_Split <- function(tib_Split, tib_merged) {

  n_Split    <- nrow(tib_Split)
  n_Split_mg <- nrow(tib_merged)

  loss_Split <- (n_Split - n_Split_mg) / n_Split * 100

  df <- data.frame(n_Split,
                   n_Split_mg,
                   signif(loss_Split, 3))

  df %>%
    kable(caption = 'Summary of total barcode counts',
          col.names = c('Split (iPCR)', 'Split (pDNA > 0)', 'Split loss (%)'))
}


get_barcode_summary_Split(tib_Split_EEP_Sox2_Combined_Unique_PlusUnknw, tib_Split_EEP_Sox2_merged)
get_barcode_summary_Split(tib_Split_EEP_Klf2_Combined_Unique_PlusUnknw, tib_Split_EEP_Klf2_merged)
get_barcode_summary_Split(tib_Split_EEP_Otx2_Combined_Unique_PlusUnknw, tib_Split_EEP_Otx2_merged)
get_barcode_summary_Split(tib_Split_EEP_FGF5_Combined_Unique_PlusUnknw, tib_Split_EEP_FGF5_merged)
get_barcode_summary_Split(tib_Split_EEP_Lefty1_Combined_Unique_PlusUnknw, tib_Split_EEP_Lefty1_merged)
get_barcode_summary_Split(tib_Split_EEP_Nanog_Combined_Unique_PlusUnknw, tib_Split_EEP_Nanog_merged)
get_barcode_summary_Split(tib_Split_EEP_P14_Combined_Unique_PlusUnknw, tib_Split_EEP_P14_merged)
get_barcode_summary_Split(tib_Split_EEP_Tbx3_Combined_Unique_PlusUnknw, tib_Split_EEP_Tbx3_merged)
```

Clean environment.

```{r}
remove(tib_Split_EEP_Sox2_Combined_Unique_PlusUnknw, tib_Split_EEP_Sox2_Combined)
remove(tib_Split_EEP_Klf2_Combined_Unique_PlusUnknw, tib_Split_EEP_Klf2_Combined)
remove(tib_Split_EEP_Otx2_Combined_Unique_PlusUnknw, tib_Split_EEP_Otx2_Combined)
remove(tib_Split_EEP_FGF5_Combined_Unique_PlusUnknw, tib_Split_EEP_FGF5_Combined)
remove(tib_Split_EEP_Lefty1_Combined_Unique_PlusUnknw, tib_Split_EEP_Lefty1_Combined)
remove(tib_Split_EEP_Nanog_Combined_Unique_PlusUnknw, tib_Split_EEP_Nanog_Combined)
remove(tib_Split_EEP_P14_Combined_Unique_PlusUnknw, tib_Split_EEP_P14_Combined)
remove(tib_Split_EEP_Tbx3_Combined_Unique_PlusUnknw, tib_Split_EEP_Tbx3_Combined)
```

# Save dataset

```{r save tibbles, eval = TRUE}
save(list = c('tib_Split_EEP_Sox2_merged','tib_Split_EEP_Klf2_merged','tib_Split_EEP_Otx2_merged','tib_Split_EEP_FGF5_merged','tib_Split_EEP_Lefty1_merged','tib_Split_EEP_Nanog_merged','tib_Split_EEP_P14_merged','tib_Split_EEP_Tbx3_merged'),
     file = '~/mydata/GitLab/epmatrix/data/MMA20230210_EEP_E143_DeepSeq_processed.RData')
```

```{r}
## Clean environment
remove(tib_cn_br1_Sox2,tib_expr_br1_Sox2, tib_expr_br2_Sox2,tib_expr_br3_Sox2)
remove(tib_cn_br1_Klf2,tib_expr_br1_Klf2, tib_expr_br2_Klf2,tib_expr_br3_Klf2)
remove(tib_cn_br1_Otx2,tib_expr_br1_Otx2, tib_expr_br2_Otx2,tib_expr_br3_Otx2)
remove(tib_cn_br1_FGF5,tib_expr_br1_FGF5, tib_expr_br2_FGF5,tib_expr_br3_FGF5)
remove(tib_cn_br1_Lefty1,tib_expr_br1_Lefty1, tib_expr_br2_Lefty1,tib_expr_br3_Lefty1)
remove(tib_cn_br1_Nanog,tib_expr_br1_Nanog, tib_expr_br2_Nanog,tib_expr_br3_Nanog)
remove(tib_cn_br1_P14,tib_expr_br1_P14, tib_expr_br2_P14,tib_expr_br3_P14)
remove(tib_cn_br1_Tbx3,tib_expr_br1_Tbx3, tib_expr_br2_Tbx3,tib_expr_br3_Tbx3)
```



# Calculate activities
```{r}
# min number of barcodes
min_n_bc <- 5
# min copy number
min_pdna <- 8

```


```{r Process function}
#adapted function from auto_process function in datteRo, does not compute cooperativity, just activities based on averaging method.
auto_process_Split <- function(tib_Split, min_pdna, min_n_bc, avg_type, signal_t) {

  # sum pdna counts across replicates
  tib_Split    <- combine_cnSplit(tib_Split)

  # remove barcodes with pDNA reads < threshold
  tib_Split    <- filter_cn(tib_Split, min_pdna = min_pdna)

  # remove barcodes of elements with < threshold distinct barcodes
  tib_Split    <- filter_barcodes(tib_Split, min_n_bc = min_n_bc,  main = 'Basal')

  # compute offsets
  offsets_Split <- tib_Split %>%
    dplyr::select(contains('dna_counts')) %>%
    colSums() / 1e6
  names(offsets_Split) <- names(offsets_Split) %>%
    str_replace('_counts', '')

  message('Split sum', offsets_Split)

  # normalize to library sizes
  tib_Split    <- normalize_counts(tib_Split, offsets_Split)

  # compute activities and summarize barcodes
  tib_Split    <- compute_activities(tib_Split)

  # summarize activities (weighted average)
  tib_Split    <- summarize_barcodesNCS(tib_Split, type = avg_type)

  #if(missing(signal_t)) {

    # model combinatorial activity distributions
    #signal_t     <- learn_min_activity(tib_comb)

  #}

  # compute cooperativity
  
  
  #tib_comb_all <- compute_cooperativity(tib_comb, tib_basal, remove_inactive = FALSE)

  # add metadata
  #tib_basal    <- add_metadata(tib_basal, tib_dat = tib_dat, type = 'basal')
  tib_Split     <- add_metadataSplit(tib_Split, tib_dat = tib_dat, type = 'comb')
  #tib_comb_all <- add_metadata(tib_comb_all, tib_dat = tib_dat, type = 'comb')

  # return list of tibs
  #return(list('basal'    = tib_basal,
  #            'comb'     = tib_comb,
  #            'comb_all' = tib_comb_all,
  #            'signal_t' = signal_t))
return(tib_Split)
}

combine_cnSplit <- function(tib) {

  # detect number of replicates based on colnames (diagnostic)
  n_repl <- dplyr::select(tib, starts_with('pdna_counts_')) %>%
    ncol

  message('Detected ', n_repl, ' replicates')

  # summarize pDNA counts across replicates
  tib %>%
    mutate(pdna_counts = rowSums(dplyr::select(., starts_with('pdna_counts_')))) %>%
    dplyr::select(-starts_with('pdna_counts_')) %>%
    dplyr::select(barcode:id, pdna_counts, starts_with('cdna_counts_'),library,Promoter,prefix1, prefix2, suffix1, suffix2)

}
add_metadataSplit <- function(tib, tib_dat, type = 'basal') {

  # validate args
  stopifnot(type %in% c('basal', 'comb'))
  if(type == 'basal') {
  }
  else {
    tib <- tib %>%
      mutate(id1   = paste0(frag1, strand1),
             id2   = paste0(frag2, strand2),
             id    = paste0(id1, id2),
             class = paste0(prefix1, prefix2)) %>%
      # add genomic coordinates
      #left_join(tib_dat, by = c('frag1' = 'frag')) %>%
      #dplyr::rename(seqnames1 = seqnames, start1 = start, end1 = end) %>%
      #left_join(tib_dat, by = c('frag2' = 'frag')) %>%
      #dplyr::rename(seqnames2 = seqnames, start2 = start, end2 = end) %>%
      # group, trick essentially rowwise()
      group_by(id1, id2,) %>%
      # distances for trans pairs is set to Inf
      #mutate(dist = ifelse(seqnames1 == seqnames2,
      #                     min(abs(end1 - start2), abs(end2 - start1)), Inf)) %>%
      # reorder to final shape
      dplyr::select(id, class, Promoter,
                    id1, frag1, strand1, prefix1, suffix1,
                    id2, frag2, strand2, prefix2, suffix2,
                    starts_with('activ'), starts_with('boost'),library) %>%
      ungroup
        }

  tib
}

summarize_barcodesNCS <- function(tib, type = 'wa') {

  # stop if type not 'wa'/'avg'
  if(!type %in% c('wa', 'avg'))
    stop('type should be equal to wa for a weighted average, or to avg for an arithmetic average')

  # aux function that works on a nested tibble ($data slot)
  get_summarized_vals <- function(dat, type) {

    # extract pDNA counts
    pdna_counts <- dat[['pdna_counts']]
    denom       <- sum(pdna_counts)

    # extract activity values
    dat <- dat %>%
      dplyr::select(starts_with('activity_'))

    if(type == 'wa') {

      # summarize barcode activities using a weighted average (w = pdna counts)
      vec_summarized <- colSums(dat * pdna_counts) / denom

    }

    else {

      # summarize barcode activities using an arithmetic average
      vec_summarized <- colMeans(dat)

    }

    # take geom mean of activities
    vec_summarized[['activity_all']] <- prod(vec_summarized) ^ (1 / length(vec_summarized))

    return(vec_summarized)
  }

  # nest input tibble by element identity
  tib_nested <- tib %>%
    group_by(id, frag1, strand1, frag2, strand2, library, Promoter, prefix1, prefix2, suffix1, suffix2) %>%
    nest()

  tib_summarized <- purrr::map(tib_nested$data, get_summarized_vals, type = type) %>%
    do.call(what = 'rbind') %>%
    as_data_frame()

  tib_nested %>%
    # drop data slot = unnest
    dplyr::select(-data) %>%
    bind_cols(tib_summarized)
}

```

```{r}


#compute activities 
tib_Split_EEP_Sox2_Activities         <- auto_process_Split(tib_Split = tib_Split_EEP_Sox2_merged,
                                              min_pdna  = min_pdna, 
                                              min_n_bc  = min_n_bc, 
                                              avg_type  = 'avg')

tib_Split_EEP_Klf2_Activities         <- auto_process_Split(tib_Split = tib_Split_EEP_Klf2_merged,
                                              min_pdna  = min_pdna, 
                                              min_n_bc  = min_n_bc, 
                                              avg_type  = 'avg')

tib_Split_EEP_Otx2_Activities         <- auto_process_Split(tib_Split = tib_Split_EEP_Otx2_merged,
                                              min_pdna  = min_pdna, 
                                              min_n_bc  = min_n_bc, 
                                              avg_type  = 'avg')

tib_Split_EEP_FGF5_Activities         <- auto_process_Split(tib_Split = tib_Split_EEP_FGF5_merged,
                                              min_pdna  = min_pdna, 
                                              min_n_bc  = min_n_bc, 
                                              avg_type  = 'avg')

tib_Split_EEP_Lefty1_Activities         <- auto_process_Split(tib_Split = tib_Split_EEP_Lefty1_merged,
                                              min_pdna  = min_pdna, 
                                              min_n_bc  = min_n_bc, 
                                              avg_type  = 'avg')

tib_Split_EEP_Nanog_Activities         <- auto_process_Split(tib_Split = tib_Split_EEP_Nanog_merged,
                                              min_pdna  = min_pdna, 
                                              min_n_bc  = min_n_bc, 
                                              avg_type  = 'avg')

tib_Split_EEP_P14_Activities         <- auto_process_Split(tib_Split = tib_Split_EEP_P14_merged,
                                              min_pdna  = min_pdna, 
                                              min_n_bc  = min_n_bc, 
                                              avg_type  = 'avg')

tib_Split_EEP_Tbx3_Activities         <- auto_process_Split(tib_Split = tib_Split_EEP_Tbx3_merged,
                                              min_pdna  = min_pdna, 
                                              min_n_bc  = min_n_bc, 
                                              avg_type  = 'avg')



```

```{r}
# Remove spike-in libs
tib_Split_EEP_Sox2_Activities%>%
  filter(Promoter%in%c("SP_Zfp822", "SP_Klf2"))%>% select(Promoter, activity_all)

tib_Split_EEP_Klf2_Activities%>%
  filter(Promoter%in%c("SP_Zfp822", "SP_Klf2"))%>% select(Promoter, activity_all)

tib_Split_EEP_Otx2_Activities%>%
  filter(Promoter%in%c("SP_Zfp822", "SP_Klf2"))%>% select(Promoter, activity_all)

tib_Split_EEP_FGF5_Activities%>%
  filter(Promoter%in%c("SP_Zfp822", "SP_Klf2"))%>% select(Promoter, activity_all)

tib_Split_EEP_Lefty1_Activities%>%
  filter(Promoter%in%c("SP_Zfp822", "SP_Klf2"))%>% select(Promoter, activity_all)

tib_Split_EEP_Nanog_Activities%>%
  filter(Promoter%in%c("SP_Zfp822", "SP_Klf2"))%>% select(Promoter, activity_all)

tib_Split_EEP_P14_Activities%>%
  filter(Promoter%in%c("SP_Zfp822", "SP_Klf2"))%>% select(Promoter, activity_all)

tib_Split_EEP_Tbx3_Activities%>%
  filter(Promoter%in%c("SP_Zfp822", "SP_Klf2"))%>% select(Promoter, activity_all)
```



# Calculate boost index



```{r}
# Calc baseline per promoter

# Sox2
NormEEP_Sox2<-tib_Split_EEP_Sox2_Activities %>%
      mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2, "Rest"))%>%
    #filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="Controls")%>%
  group_by(Promoter)%>%
  summarise(Baseline=median(activity_all), StDevBaseline= sd(activity_all))

tib_Split_EEP_Sox2_Boost <- tib_Split_EEP_Sox2_Activities %>% left_join(NormEEP_Sox2, by = "Promoter") %>%
  mutate(boost=log2(activity_all/Baseline))

# Klf2
NormEEP_Klf2<-tib_Split_EEP_Klf2_Activities %>%
      mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2, "Rest"))%>%
    #filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="Controls")%>%
  group_by(Promoter)%>%
  summarise(Baseline=median(activity_all), StDevBaseline= sd(activity_all))

tib_Split_EEP_Klf2_Boost <- tib_Split_EEP_Klf2_Activities %>% left_join(NormEEP_Klf2, by = "Promoter") %>%
  mutate(boost=log2(activity_all/Baseline))

# Otx2
NormEEP_Otx2<-tib_Split_EEP_Otx2_Activities %>%
      mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2, "Rest"))%>%
    #filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="Controls")%>%
  group_by(Promoter)%>%
  summarise(Baseline=median(activity_all), StDevBaseline= sd(activity_all))

tib_Split_EEP_Otx2_Boost <- tib_Split_EEP_Otx2_Activities %>% left_join(NormEEP_Otx2, by = "Promoter") %>%
  mutate(boost=log2(activity_all/Baseline))

# FGF5
NormEEP_FGF5<-tib_Split_EEP_FGF5_Activities %>%
      mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2, "Rest"))%>%
    #filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="Controls")%>%
  group_by(Promoter)%>%
  summarise(Baseline=median(activity_all), StDevBaseline= sd(activity_all))

tib_Split_EEP_FGF5_Boost <- tib_Split_EEP_FGF5_Activities %>% left_join(NormEEP_FGF5, by = "Promoter") %>%
  mutate(boost=log2(activity_all/Baseline))

# Lefty1
NormEEP_Lefty1<-tib_Split_EEP_Lefty1_Activities %>%
      mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2, "Rest"))%>%
    #filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="Controls")%>%
  group_by(Promoter)%>%
  summarise(Baseline=median(activity_all), StDevBaseline= sd(activity_all))

tib_Split_EEP_Lefty1_Boost <- tib_Split_EEP_Lefty1_Activities %>% left_join(NormEEP_Lefty1, by = "Promoter") %>%
  mutate(boost=log2(activity_all/Baseline))

# Nanog
NormEEP_Nanog<-tib_Split_EEP_Nanog_Activities %>%
      mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2, "Rest"))%>%
    #filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="Controls")%>%
  group_by(Promoter)%>%
  summarise(Baseline=median(activity_all), StDevBaseline= sd(activity_all))

tib_Split_EEP_Nanog_Boost <- tib_Split_EEP_Nanog_Activities %>% left_join(NormEEP_Nanog, by = "Promoter") %>%
  mutate(boost=log2(activity_all/Baseline))

# P14
NormEEP_P14<-tib_Split_EEP_P14_Activities %>%
      mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2, "Rest"))%>%
    #filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="Controls")%>%
  group_by(Promoter)%>%
  summarise(Baseline=median(activity_all), StDevBaseline= sd(activity_all))

tib_Split_EEP_P14_Boost <- tib_Split_EEP_P14_Activities %>% left_join(NormEEP_P14, by = "Promoter") %>%
  mutate(boost=log2(activity_all/Baseline))

# Tbx3
NormEEP_Tbx3<-tib_Split_EEP_Tbx3_Activities %>%
      mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2, "Rest"))%>%
    #filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="Controls")%>%
  group_by(Promoter)%>%
  summarise(Baseline=median(activity_all), StDevBaseline= sd(activity_all))

tib_Split_EEP_Tbx3_Boost <- tib_Split_EEP_Tbx3_Activities %>% left_join(NormEEP_Tbx3, by = "Promoter") %>%
  mutate(boost=log2(activity_all/Baseline))

```


```{r, fig.width = 10, fig.height = 10}

tib_Split_EEP_Sox2_Boost %>%
  filter(Promoter=="P1_SOX2")%>%
  mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,"Rest"))%>%
  ggplot(aes(cluster, boost, fill=cluster))+
  geom_boxplot()+
  facet_wrap(~Promoter)+
  labs(title = "Sox2 EEP Boost")
```


# Calculate boost per single enhancer (enhacer-control combinations)


```{r}
# Calc baseline per single element
NormEEPSingle1_Sox2<-tib_Split_EEP_Sox2_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix2=="Controls")%>%
    group_by(frag1,Promoter)%>%
  summarise(Baseline1=median(activity_all), n=n(), StDevBaseline1= sd(activity_all)) %>%
  filter(n>5)

NormEEPSingle2_Sox2<-tib_Split_EEP_Sox2_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix1=="Controls")%>%
  group_by(frag2,Promoter)%>%
  summarise(Baseline2=median(activity_all), n=n(), StDevBaseline2= sd(activity_all)) %>%
  filter(n>5)
```

```{r}
# Calc baseline per single element
NormEEPSingle1_Klf2<-tib_Split_EEP_Klf2_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix2=="Controls")%>%
    group_by(frag1,Promoter)%>%
  summarise(Baseline1=median(activity_all), n=n(), StDevBaseline1= sd(activity_all)) %>%
  filter(n>5)

NormEEPSingle2_Klf2<-tib_Split_EEP_Klf2_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix1=="Controls")%>%
  group_by(frag2,Promoter)%>%
  summarise(Baseline2=median(activity_all), n=n(), StDevBaseline2= sd(activity_all)) %>%
  filter(n>5)
```

```{r}
# Calc baseline per single element
NormEEPSingle1_Otx2<-tib_Split_EEP_Otx2_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix2=="Controls")%>%
    group_by(frag1,Promoter)%>%
  summarise(Baseline1=median(activity_all), n=n(), StDevBaseline1= sd(activity_all)) %>%
  filter(n>5)

NormEEPSingle2_Otx2<-tib_Split_EEP_Otx2_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix1=="Controls")%>%
  group_by(frag2,Promoter)%>%
  summarise(Baseline2=median(activity_all), n=n(), StDevBaseline2= sd(activity_all)) %>%
  filter(n>5)
```

```{r}
# Calc baseline per single element
NormEEPSingle1_FGF5<-tib_Split_EEP_FGF5_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix2=="Controls")%>%
    group_by(frag1,Promoter)%>%
  summarise(Baseline1=median(activity_all), n=n(), StDevBaseline1= sd(activity_all)) %>%
  filter(n>5)

NormEEPSingle2_FGF5<-tib_Split_EEP_FGF5_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix1=="Controls")%>%
  group_by(frag2,Promoter)%>%
  summarise(Baseline2=median(activity_all), n=n(), StDevBaseline2= sd(activity_all)) %>%
  filter(n>5)
```

```{r}
# Calc baseline per single element
NormEEPSingle1_Lefty1<-tib_Split_EEP_Lefty1_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix2=="Controls")%>%
    group_by(frag1,Promoter)%>%
  summarise(Baseline1=median(activity_all), n=n(), StDevBaseline1= sd(activity_all)) %>%
  filter(n>5)

NormEEPSingle2_Lefty1<-tib_Split_EEP_Lefty1_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix1=="Controls")%>%
  group_by(frag2,Promoter)%>%
  summarise(Baseline2=median(activity_all), n=n(), StDevBaseline2= sd(activity_all)) %>%
  filter(n>5)
```

```{r}
# Calc baseline per single element
NormEEPSingle1_Nanog<-tib_Split_EEP_Nanog_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix2=="Controls")%>%
    group_by(frag1,Promoter)%>%
  summarise(Baseline1=median(activity_all), n=n(), StDevBaseline1= sd(activity_all)) %>%
  filter(n>5)

NormEEPSingle2_Nanog<-tib_Split_EEP_Nanog_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix1=="Controls")%>%
  group_by(frag2,Promoter)%>%
  summarise(Baseline2=median(activity_all), n=n(), StDevBaseline2= sd(activity_all)) %>%
  filter(n>5)
```

```{r}
# Calc baseline per single element
NormEEPSingle1_P14<-tib_Split_EEP_P14_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix2=="Controls")%>%
    group_by(frag1,Promoter)%>%
  summarise(Baseline1=median(activity_all), n=n(), StDevBaseline1= sd(activity_all)) %>%
  filter(n>5)

NormEEPSingle2_P14<-tib_Split_EEP_P14_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix1=="Controls")%>%
  group_by(frag2,Promoter)%>%
  summarise(Baseline2=median(activity_all), n=n(), StDevBaseline2= sd(activity_all)) %>%
  filter(n>5)
```

```{r}
# Calc baseline per single element
NormEEPSingle1_Tbx3<-tib_Split_EEP_Tbx3_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix2=="Controls")%>%
    group_by(frag1,Promoter)%>%
  summarise(Baseline1=median(activity_all), n=n(), StDevBaseline1= sd(activity_all)) %>%
  filter(n>5)

NormEEPSingle2_Tbx3<-tib_Split_EEP_Tbx3_Boost %>%
     mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!(frag1==frag2 & strand1==strand2))%>%
  filter(cluster=="SingleControls")%>%
  filter(suffix1=="Controls")%>%
  group_by(frag2,Promoter)%>%
  summarise(Baseline2=median(activity_all), n=n(), StDevBaseline2= sd(activity_all)) %>%
  filter(n>5)
```


## Calculate single enhancer effects


```{r}
tib_Split_EEP_Sox2_normDual<-tib_Split_EEP_Sox2_Boost%>%
  mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!((frag1==frag2) & (strand1==strand2)))%>%
  left_join(.,NormEEPSingle1_Sox2%>%
  ungroup(), by=c("frag1", "Promoter"))%>%
  left_join(.,NormEEPSingle2_Sox2%>%
  ungroup(), by=c("frag2", "Promoter"))%>%
  #mutate(AdditiveCoop= log2((Activity-Baseline)/(Baseline1+Baseline2-(2*Baseline))))%>%
  mutate(AdditiveCoop= log2((activity_all)/(Baseline1+Baseline2-Baseline)))%>%
  mutate(Additivebaseline= Baseline1+Baseline2-Baseline)%>%
 mutate(MiltipCoop= log2((activity_all/Baseline)/(Baseline1*Baseline2/(Baseline*Baseline))))%>%
  mutate(Boost1= log2((Baseline1)/(Baseline)))%>%
  mutate(Boost2= log2((Baseline2)/(Baseline)))%>%
  mutate(Coop= boost-Boost1-Boost2)%>%
  mutate(R1= (Baseline1)/(Baseline))%>%
  mutate(R2= (Baseline2)/(Baseline))%>%
  mutate(Coop2= log2(boost-R1-R2))


```

```{r}
tib_Split_EEP_Klf2_normDual<-tib_Split_EEP_Klf2_Boost%>%
  mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!((frag1==frag2) & (strand1==strand2)))%>%
  left_join(.,NormEEPSingle1_Klf2%>%
  ungroup(), by=c("frag1", "Promoter"))%>%
  left_join(.,NormEEPSingle2_Klf2%>%
  ungroup(), by=c("frag2", "Promoter"))%>%
  #mutate(AdditiveCoop= log2((Activity-Baseline)/(Baseline1+Baseline2-(2*Baseline))))%>%
  mutate(AdditiveCoop= log2((activity_all)/(Baseline1+Baseline2-Baseline)))%>%
  mutate(Additivebaseline= Baseline1+Baseline2-Baseline)%>%
 mutate(MiltipCoop= log2((activity_all/Baseline)/(Baseline1*Baseline2/(Baseline*Baseline))))%>%
  mutate(Boost1= log2((Baseline1)/(Baseline)))%>%
  mutate(Boost2= log2((Baseline2)/(Baseline)))%>%
  mutate(Coop= boost-Boost1-Boost2)%>%
  mutate(R1= (Baseline1)/(Baseline))%>%
  mutate(R2= (Baseline2)/(Baseline))%>%
  mutate(Coop2= log2(boost-R1-R2))


```

```{r}
tib_Split_EEP_Otx2_normDual<-tib_Split_EEP_Otx2_Boost%>%
  mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!((frag1==frag2) & (strand1==strand2)))%>%
  left_join(.,NormEEPSingle1_Otx2%>%
  ungroup(), by=c("frag1", "Promoter"))%>%
  left_join(.,NormEEPSingle2_Otx2%>%
  ungroup(), by=c("frag2", "Promoter"))%>%
  #mutate(AdditiveCoop= log2((Activity-Baseline)/(Baseline1+Baseline2-(2*Baseline))))%>%
  mutate(AdditiveCoop= log2((activity_all)/(Baseline1+Baseline2-Baseline)))%>%
  mutate(Additivebaseline= Baseline1+Baseline2-Baseline)%>%
 mutate(MiltipCoop= log2((activity_all/Baseline)/(Baseline1*Baseline2/(Baseline*Baseline))))%>%
  mutate(Boost1= log2((Baseline1)/(Baseline)))%>%
  mutate(Boost2= log2((Baseline2)/(Baseline)))%>%
  mutate(Coop= boost-Boost1-Boost2)%>%
  mutate(R1= (Baseline1)/(Baseline))%>%
  mutate(R2= (Baseline2)/(Baseline))%>%
  mutate(Coop2= log2(boost-R1-R2))


```

```{r}
tib_Split_EEP_FGF5_normDual<-tib_Split_EEP_FGF5_Boost%>%
  mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!((frag1==frag2) & (strand1==strand2)))%>%
  left_join(.,NormEEPSingle1_FGF5%>%
  ungroup(), by=c("frag1", "Promoter"))%>%
  left_join(.,NormEEPSingle2_FGF5%>%
  ungroup(), by=c("frag2", "Promoter"))%>%
  #mutate(AdditiveCoop= log2((Activity-Baseline)/(Baseline1+Baseline2-(2*Baseline))))%>%
  mutate(AdditiveCoop= log2((activity_all)/(Baseline1+Baseline2-Baseline)))%>%
  mutate(Additivebaseline= Baseline1+Baseline2-Baseline)%>%
 mutate(MiltipCoop= log2((activity_all/Baseline)/(Baseline1*Baseline2/(Baseline*Baseline))))%>%
  mutate(Boost1= log2((Baseline1)/(Baseline)))%>%
  mutate(Boost2= log2((Baseline2)/(Baseline)))%>%
  mutate(Coop= boost-Boost1-Boost2)%>%
  mutate(R1= (Baseline1)/(Baseline))%>%
  mutate(R2= (Baseline2)/(Baseline))%>%
  mutate(Coop2= log2(boost-R1-R2))


```

```{r}
tib_Split_EEP_Lefty1_normDual<-tib_Split_EEP_Lefty1_Boost%>%
  mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!((frag1==frag2) & (strand1==strand2)))%>%
  left_join(.,NormEEPSingle1_Lefty1%>%
  ungroup(), by=c("frag1", "Promoter"))%>%
  left_join(.,NormEEPSingle2_Lefty1%>%
  ungroup(), by=c("frag2", "Promoter"))%>%
  #mutate(AdditiveCoop= log2((Activity-Baseline)/(Baseline1+Baseline2-(2*Baseline))))%>%
  mutate(AdditiveCoop= log2((activity_all)/(Baseline1+Baseline2-Baseline)))%>%
  mutate(Additivebaseline= Baseline1+Baseline2-Baseline)%>%
 mutate(MiltipCoop= log2((activity_all/Baseline)/(Baseline1*Baseline2/(Baseline*Baseline))))%>%
  mutate(Boost1= log2((Baseline1)/(Baseline)))%>%
  mutate(Boost2= log2((Baseline2)/(Baseline)))%>%
  mutate(Coop= boost-Boost1-Boost2)%>%
  mutate(R1= (Baseline1)/(Baseline))%>%
  mutate(R2= (Baseline2)/(Baseline))%>%
  mutate(Coop2= log2(boost-R1-R2))


```

```{r}
tib_Split_EEP_Nanog_normDual<-tib_Split_EEP_Nanog_Boost%>%
  mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!((frag1==frag2) & (strand1==strand2)))%>%
  left_join(.,NormEEPSingle1_Nanog%>%
  ungroup(), by=c("frag1", "Promoter"))%>%
  left_join(.,NormEEPSingle2_Nanog%>%
  ungroup(), by=c("frag2", "Promoter"))%>%
  #mutate(AdditiveCoop= log2((Activity-Baseline)/(Baseline1+Baseline2-(2*Baseline))))%>%
  mutate(AdditiveCoop= log2((activity_all)/(Baseline1+Baseline2-Baseline)))%>%
  mutate(Additivebaseline= Baseline1+Baseline2-Baseline)%>%
 mutate(MiltipCoop= log2((activity_all/Baseline)/(Baseline1*Baseline2/(Baseline*Baseline))))%>%
  mutate(Boost1= log2((Baseline1)/(Baseline)))%>%
  mutate(Boost2= log2((Baseline2)/(Baseline)))%>%
  mutate(Coop= boost-Boost1-Boost2)%>%
  mutate(R1= (Baseline1)/(Baseline))%>%
  mutate(R2= (Baseline2)/(Baseline))%>%
  mutate(Coop2= log2(boost-R1-R2))


```

```{r}
tib_Split_EEP_P14_normDual<-tib_Split_EEP_P14_Boost%>%
  mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!((frag1==frag2) & (strand1==strand2)))%>%
  left_join(.,NormEEPSingle1_P14%>%
  ungroup(), by=c("frag1", "Promoter"))%>%
  left_join(.,NormEEPSingle2_P14%>%
  ungroup(), by=c("frag2", "Promoter"))%>%
  #mutate(AdditiveCoop= log2((Activity-Baseline)/(Baseline1+Baseline2-(2*Baseline))))%>%
  mutate(AdditiveCoop= log2((activity_all)/(Baseline1+Baseline2-Baseline)))%>%
  mutate(Additivebaseline= Baseline1+Baseline2-Baseline)%>%
 mutate(MiltipCoop= log2((activity_all/Baseline)/(Baseline1*Baseline2/(Baseline*Baseline))))%>%
  mutate(Boost1= log2((Baseline1)/(Baseline)))%>%
  mutate(Boost2= log2((Baseline2)/(Baseline)))%>%
  mutate(Coop= boost-Boost1-Boost2)%>%
  mutate(R1= (Baseline1)/(Baseline))%>%
  mutate(R2= (Baseline2)/(Baseline))%>%
  mutate(Coop2= log2(boost-R1-R2))


```

```{r}
tib_Split_EEP_Tbx3_normDual<-tib_Split_EEP_Tbx3_Boost%>%
  mutate(suffix1= ifelse(grepl('shuffled',suffix1) ,"Controls", 
                             ifelse(grepl('Random',suffix1) ,"Controls", suffix1)))%>%
        mutate(suffix2= ifelse(grepl('shuffled',suffix2) ,"Controls", 
                             ifelse(grepl('Random',suffix2) ,"Controls", suffix2)))%>%
  mutate(cluster= ifelse(suffix1==suffix2 ,suffix2,
                         ifelse(suffix1== "Controls" |suffix2== "Controls" ,"SingleControls",
                         "Rest")))%>%
    filter(!((frag1==frag2) & (strand1==strand2)))%>%
  left_join(.,NormEEPSingle1_Tbx3%>%
  ungroup(), by=c("frag1", "Promoter"))%>%
  left_join(.,NormEEPSingle2_Tbx3%>%
  ungroup(), by=c("frag2", "Promoter"))%>%
  #mutate(AdditiveCoop= log2((Activity-Baseline)/(Baseline1+Baseline2-(2*Baseline))))%>%
  mutate(AdditiveCoop= log2((activity_all)/(Baseline1+Baseline2-Baseline)))%>%
  mutate(Additivebaseline= Baseline1+Baseline2-Baseline)%>%
 mutate(MiltipCoop= log2((activity_all/Baseline)/(Baseline1*Baseline2/(Baseline*Baseline))))%>%
  mutate(Boost1= log2((Baseline1)/(Baseline)))%>%
  mutate(Boost2= log2((Baseline2)/(Baseline)))%>%
  mutate(Coop= boost-Boost1-Boost2)%>%
  mutate(R1= (Baseline1)/(Baseline))%>%
  mutate(R2= (Baseline2)/(Baseline))%>%
  mutate(Coop2= log2(boost-R1-R2))


```


# Save dataset norm

```{r}
save(list = c('tib_Split_EEP_Sox2_normDual','tib_Split_EEP_Klf2_normDual','tib_Split_EEP_Otx2_normDual','tib_Split_EEP_FGF5_normDual','tib_Split_EEP_Lefty1_normDual','tib_Split_EEP_Nanog_normDual','tib_Split_EEP_P14_normDual','tib_Split_EEP_Tbx3_normDual'),
     file = '~/mydata/GitLab/epmatrix/data/MMA20230210_EEP_E143_DeepSeq_Norm.RData')
```

```{r}
## Remove Objects not needed
rm(list = ls())
```



